user_sync/connector/directory_okta.py

```diff
@@ -185,13 +172,10 @@ def iter_group_members(self, group, filter_string, extended_attributes):
                 self.logger.warning("Unable to get_group_users")
                 raise user_sync.error.AssertionException(repr(e))
             #Filtering users based all_users_filter query in config
-            if filter_string:
-                members = self.filter_user(members,filter_string)
-            for member in members:
+            for member in self.filter_users(members, filter_string):
                 profile = member.profile
                 if not profile.email:
-                    # if (last_attribute_name != None):
-                    self.logger.warn('No email attribute for login: %s', profile.login)
+                    self.logger.warning('No email attribute for login: %s', profile.login)
```

Comment:
by the way, you got this right, and ldap got it wrong - it checks for missing email attribute, but doesn't catch when email is empty.  i'll fix that. :)